clients/fides-js/src/lib/consent-types.ts

```diff
@@ -199,9 +199,16 @@ export class SaveConsentPreference {
 
   notice: PrivacyNotice;
 
-  constructor(notice: PrivacyNotice, consentPreference: UserConsentPreference) {
+  noticeHistoryId?: string;
+
+  constructor(
+    notice: PrivacyNotice,
+    consentPreference: UserConsentPreference,
+    noticeHistoryId?: string
+  ) {
     this.notice = notice;
     this.consentPreference = consentPreference;
+    this.noticeHistoryId = noticeHistoryId;
   }
```

Comment:
I know noticeHistoryId is optional. You have though updated everywhere SaveConsentPreference is used expect on `clients/fides-js/src/lib/cookie.ts` in `updateCookieFromNoticePreferences` so putting that here for your consideration.

Reply:
That's a good suggestion, thanks... I was heavily utilizing Typescript for this kind of refactor so I made these IDs optional in some places (e.g. at the final API surface) but was trying to enforce them most of the time.

I just tried making it required but immediately ran into an issue in that I can't really _guarantee_ that a translation will be found by `selectBestNoticeTranslation` so even `NoticeOverlay.tsx:168` can fail to find a valid history ID when creating a new `SaveConsentPreference`... so I need to just let it be optional in the types 👎